embedchain/embedchain.py

```diff
@@ -589,9 +603,13 @@ def chat(
         or the dry run result
         :rtype: str, if citations is False, otherwise Tuple[str,List[Tuple[str,str,str]]]
         """
-        citations = kwargs.get("citations", False)
+        if "citations" in kwargs:
+            citations = kwargs.pop("citations")
```

Comment:
Changing the arguments passed to a function inside that function is never a good idea. You can just do `get()` here and be more pythonic.

Reply:
Then how to avoid passing `citations` kwarg unnecessarily to the other functions? As it might break other functions because of unexpected kwarg.

Reply:
You can do something like this:

```python
params = {}
params['citations'] = kwargs['citations']

my_func(**params)
```

Reply:
Feel free to fix this in a follow up PR. Will merge this for now.